components/Navigation.tsx

```diff
@@ -14,9 +15,16 @@ export default function Navigation() {
     useContext(AuthContext);
 
   const [isSignOutShowing, setIsSignOutShowing] = useState(false);
+  const [isOpenNav, setIsOpenNav] = useState(false);
 
   return (
-    <div className="w-full bg-semiDarkBlue p-4 flex flex-row items-center justify-between  lg:w-20 lg:flex-col lg:h-[calc(100vh_-_2rem)] lg:rounded-lg">
+    <motion.div
+      initial="initial"
+      animate="animate"
+      className={`w-full bg-semiDarkBlue p-4 flex flex-row items-center justify-between   lg:flex-col lg:h-[calc(100vh_-_2rem)] lg:rounded-lg ${
+        isOpenNav ? "w-56" : "lg:w-20"
+      }`}
+    >
       <Link href="/">
```

Comment:
instead of manually adding new pages, do this instead

```
const menuItems = [{
href: '/',
label: 'Home',
svg: SOME_SVG
},
... add the rest over here
]
```